### PR TITLE
Pass in public hosted zone ID instead of searching for it

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request modifies this module, removing the `domain` input variable and adding a `public_zone_id` input variable.  Internally, it is modified to use the public hosted zone ID directly instead of searching for it via the domain.

## 💭 Motivation and Context

In cisagov/cool-dns-cyber.dhs.gov#8, @felddy pointed out that I shouldn't need to modify that repository if I instead modify this module to pass in the public hosted zone ID instead of searching for it via the domain.  This is a cleaner solution.

## 🧪 Testing

I was able to successfully deploy cisagov/cool-sharedservices-openvpn#10 using these changes.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
LGTM! 👍

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

**Comment by jsf9k** [comment]
I'll wait for @felddy to approve these changes before I merge, since he asked for them.

**Comment by dav3r** [comment]
FYI, I have some changes ready to go for #17, but I will put them in a separate PR.

**Review by felddy — APPROVED** [review]
Gentleman and a scholar!  🎩

**Pull request merged**
